ext/date/tests/009.phpt

```diff
@@ -10,13 +10,21 @@ date_default_timezone_set('Asia/Jerusalem');
 
 $t = mktime(0,0,0, 6, 27, 2006);
 
-var_dump(strftime(""));
+try {
+    var_dump(strftime(""));
+} catch (\ValueError $e) {
+    echo $e->getMessage() . \PHP_EOL;
+}
```

Comment:
I am not keen on this change, as this is a BC break. Previously wrong values just returned `false`, which is something that `strtotime` and the older date functions have done for decades. Upgrading them to throw an exception is not a great thing to do here. It's going to cause BC breaks, even if people already carefully checked the return values. As most input to date functions will come from either user input or a database, I don't believe this is a worthy place to introduce breaks.

Instead, we should point people to the DateTime classes.

Reply:
Are you trying to promote the use of the DateTim(Immutable) class for this specific case, or more generally?

From what I've understood the DateTime classes also return false and don't throw, or would you be more inclined for the objects to throw and be able to drop the false return?